docs/source/builtin_types.rst

```diff
@@ -38,3 +38,8 @@ though they are similar to abstract base classes defined in
 :py:mod:`collections.abc` (formerly ``collections``), they are not identical. In
 particular, prior to Python 3.9, the built-in collection type objects do not
 support indexing.
+
+In Python 3.9 and later, built-in collection type objects support indexing. This
+means that you can use built-in classes or those from :py:mod:`collections.abc`
+instead of importing from :py:mod:`typing`. See :ref:`generic-builtins` for more
+details.
```

Comment:
Since after a while a significant fraction (eventually a majority) of users will be using Python 3.9+, at some point we should make the 3.9+ way of doing things more prominent. Initially we could add another column such as "Type (Python 3.9)", which would have examples like `list[str]`.

No need to do this in this PR, though.